python/lsst/ap/association/testApdb.py

```diff
@@ -0,0 +1,575 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+
+"""Standalone pipelinetask to populate an APDB with simulated data.
+"""
+
+__all__ = ("TestApdbConfig",
+           "TestApdbTask",
+           )
+
+
+import numpy as np
+import pandas as pd
+
+from lsst.daf.base import DateTime
+import lsst.dax.apdb as daxApdb
+import lsst.geom
+from lsst.meas.base import DetectorVisitIdGeneratorConfig, IdGenerator
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from lsst.ap.association.loadDiaCatalogs import LoadDiaCatalogsTask
+from lsst.ap.association.diaPipe import DiaPipelineTask
+from lsst.ap.association.utils import (
+    convertTableToSdmSchema,
+    readSchemaFromApdb,
+    column_dtype,
+    make_empty_catalog,
+)
+import lsst.sphgeom
+
+
+class TestApdbConnections(
+        pipeBase.PipelineTaskConnections,
+        dimensions=("instrument", "visit", "detector")):
+    """Butler connections for TestApdbTask.
+    """
+
+
+class TestApdbConfig(pipeBase.PipelineTaskConfig,
+                     pipelineConnections=TestApdbConnections):
+    """Config for TestApdbTask.
+    """
+    apdb_config_url = pexConfig.Field(
+        dtype=str,
+        default=None,
+        optional=False,
+        doc="A config file specifying the APDB and its connection parameters, "
+            "typically written by the apdb-cli command-line utility. "
+            "The database must already be initialized.",
+    )
+    survey_area = pexConfig.Field(
+        dtype=float,
+        default=10,
+        doc="Area (in degrees) of the simulated survey",
+    )
+    fov = pexConfig.Field(
+        dtype=float,
+        default=9.6,
+        doc="Field of view of the camera, in square degrees.",
+    )
+    stellar_density = pexConfig.Field(
+        dtype=float,
+        default=1750,
+        doc="Average number of real transient and variable objects"
+            " detected per square degree. For this simulation, these will"
+            " always be detected, and detected in the same location."
+            "The default is chosen such that:"
+            " Density x Rubin Fov (9.6) x # of visits per night (~600) ~ 10M",
+    )
+    false_positive_ratio = pexConfig.Field(
+        dtype=float,
+        default=4,
+        doc="Average ratio of false detections to real sources."
+            "These will be detected in random locations.",
```

Comment:
LDM-141 expects false positives to be 50% of real alerts, if the actual number is 400%, do we need to update LDM-141?

Reply:
Good question, since we are nowhere close to 2:1 real to bogus sources currently. This ratio is over a year old, though, and has not been re-evaluated from ComCam data.